partition/bootloader.go

```diff
@@ -119,7 +119,23 @@ func ForceBootloader(booloader Bootloader) {
 
 // MarkBootSuccessful marks the current boot as successful. This means
 // that snappy will consider this combination of kernel/os a valid
-// target for rollback
+// target for rollback.
+//
+// The states that a boot goes through are the following:
+// - By default snap_mode is "" in which case the bootloader loads
+//   the snap_core and snap_kernel.
+// - On a refresh of core/kernel snapd will set snap_mode=try and
+//   will also set snap_try_{core,kernel} to the core/kernel that
+//   will be tried next.
+// - On reboot the bootloader will inspect the snap_mode and if the
+//   mode is set to "try" it will set "snap_mode=trying" and then
+//   try to boot the snap_try_{core,kernel}".
+// - On a (successful) boot snapd will set snap_mode="" (default)
+//   and also snap_{core,kernel} is set to the snap_try_{core,kernel}
```

Comment:
On a successful boot snapd resets `snap_mode` to `""` and copies `snap_try_{core,kernel}` to `snap_{core,kernel}`.

Question, do we reset `snap_try_{core,kernel}` at this stage or are they left as-is?